### ISSUE TITLE
Bug: Game Crashes upon reaching the Hall of Fame

### ISSUE THREAD

**Issue by Greenknight9000** [jojobear13/shinpokered#280]
**Review the [Compatibility Notes](https://github.com/jojobear13/shinpokered/#compatibility-notes)**
In particular, not all emulators or non-official hardware solutions are supported due to bugs induced from lack of accuracy. Proper bugs should be able to be replicated on official hardware or, to a lesser extent, the latest version of the BGB emulator.

**What emulator and/or hardware are you are using?**
BGB

**Which patch are you using?**
Version 1.23.15, compiled from Git

**What ROM base are you using?**
Pokémon Red

**How are you utilizing the patch?**
I compiled the ROM myself via Cygwin64

**What version of this project are you using?**
V1.23.15M

**What is the problem you encountered?**
Upon reaching the Hall of Fame, when the Hall of Fame is set to display the player after showing the Pokémon, the game hangs on a white screen.

**Expected behavior**
The player is shown along with their total playtime, the game then continues to credits

**Is it possible to reproduce the problem?**
If so, please list the exact steps taken to reproduce the erroneous behavior.
Example:
1: Beat the Champion Battle
2: Let the Hall of Fame progress. When it reaches the player, it should hang

**Screenshots/Video**

https://github.com/jojobear13/shinpokered/assets/91160044/7bae44f1-a506-43d2-b3a6-775fdec67c25

**SAV file or Save States**
[PokeRed Save and SaveState.zip](https://github.com/jojobear13/shinpokered/files/12690197/PokeRed.Save.and.SaveState.zip)

**Additional context**
I have tried compiling the game multiple times, even tried the default patches provided, yet they still produce a crash. I initially thought it was due to using modified sprites, so I just used the default SW sprites, yet they still crash; I tried using the original backsprites, and it still crashed.

**Comment by epman100** [comment]
> *
> 
> **Which patch are you using?** Version 1.23.15, compiled from Git
> 

the game did not crash on my end on reaching the hall of fame.  played shin pokemon red v1.23.15 in BGB v1.5.10 on my win10 pc and used the ips patch file, not by compilation

> **Additional context** I have tried compiling the game multiple times, even tried the default patches provided, yet they still produce a crash. I initially thought it was due to using modified sprites, so I just used the default SW sprites, yet they still crash; I tried using the original backsprites, and it still crashed.

maybe that's why it crashed for you but not for me.

I used the 1.23.15 ips patch files directly from [here](https://github.com/jojobear13/shinpokered/tree/master_dev/patches_and_info) (when the 1.23.15 patches were still available but have recently been replaced by the 1.24.0 versions) and **not** by compiling the game

**Comment by jojobear13** [comment]
Give me 24 to 48 hours to look into this. My laptop has a fried power regulator and I have a new mobo on overnight order.

**Comment by jojobear13** [comment]
Nice catch [Greenknight9000](https://github.com/Greenknight9000). What you compiled was later code closer to v1.24.0, and it looks like I made a goof. I had moved hall_of_fame.asm to a different bank to save space, but I missed that there is a call to save the hall of fame data which relies on being in the same bank. Oops. I pushed a commit to the master_dev and lite_dev branches just now to fix that. I moved  hall_of_fame.asm back into bank 1C and swapped in_game_trades.asm to a different bank (which by a quick audit is self-contained).

I have attached a patch, a save file, and a save state that reflect the commits I just made. They should get through to the credits now.
[fixed files.zip](https://github.com/jojobear13/shinpokered/files/12707951/fixed.files.zip)
